src/libchdr_chd.c

```diff
@@ -1633,17 +1657,12 @@ CHD_EXPORT chd_error chd_open_file(core_file *file, int mode, chd_file *parent,
 
 CHD_EXPORT chd_error chd_precache(chd_file *chd)
 {
-#ifdef _MSC_VER
-	size_t size, count;
-#else
-	ssize_t size, count;
-#endif
+	INT64 count;
 
 	if (chd->file_cache == NULL)
 	{
-		core_fseek(chd->file, 0, SEEK_END);
-		size = core_ftell(chd->file);
-		if (size <= 0)
+		UINT64 size = core_fsize(chd->file);
```

Comment:
If support is required for older compilers lacking online variable declaration, move this declaration to the start of the function.